python/lsst/ap/association/association.py

```diff
@@ -135,53 +143,64 @@ def run(self, dia_sources, exposure):
             bbox=bbox,
             wcs=wcs,)
 
-        dia_collection = self.level1_db.load(expMd)
+        dia_objects = self.level1_db.load_dia_objects(expMd)
 
-        association_result = self.associate_sources(dia_collection,
-                                                    dia_sources)
+        association_result = self.associate_sources(dia_objects, dia_sources)
+        updated_obj_ids = association_result.updated_obj_ids
 
-        dia_collection = association_result.dia_collection
-        updated_obj_ids = association_result.updated_ids
+        # Store newly associated DIASources.
+        self.level1_db.store_dia_sources(dia_sources, updated_obj_ids)
+        # Update previously existing DIAObjects with the information from their
+        # newly association DIASources.
+        self.update_dia_objects(updated_obj_ids)
 
-        dia_collection.update_dia_objects()
+    @pipeBase.timeMethod
+    def update_dia_objects(self, updated_obj_ids):
 
-        self.level1_db.store_updated(dia_collection, updated_obj_ids)
+        for obj_id in updated_obj_ids:
+            tmp_dia_object_record = afwTable.SourceTable.makeRecord(
+                afwTable.SourceTable.make(
+                    self.level1_db.get_dia_object_schema()))
+            tmp_dia_object_record.set('id', obj_id)
```

Comment:
Since I'm going to be updating the values of the DIAObject anyway it makes more sense to just create a new one, fill it, and store it in the DB rather than indexing the DIAObject catalog and have a similar one at a time loop.

Reply:
Fine, but making the new record seems a lot more complicated than I would expect.

Also, I missed it the first time around, but consider storing the records in a list and calling `store_dia_objects` outside the loop. That should minimize the overhead from the database.

Reply:
I've changed this up a bit based on a comment you had later. I now create a catalog and update the values of the objects as I go. This is related to a comment you made later about doing single inserts into the database. I've changed AssociationDBSqlite._store_record to _store_catalog to do the multiple inserts in a straight forward way.